### ISSUE TITLE
Incorrect type checking for objects with call properties

### ISSUE THREAD

**Issue by samwgoldman** [facebook/flow#458]
Hopefully the following test cases are clear. There seem to be a few different issues here.

```
/* @flow */

type Test = {
  (x: string): number;
  foo: string;
}

function ex1(): Test {
  // not ok (property foo not found) -- flow reports expected error
  return function(x: string) {
    return parseInt(x)
  }
}

function ex2(): Test {
  // not ok (function return type is wrong) -- flow reports expected error
  var f = function(x) { }
  f.foo = "foo";
  return f;
}

function ex3(): Test {
  // ok -- flow reports unexpected error (too few arguments)
  var f = function() { return 1 }
  f.foo = "foo";
  return f;
}

function ex4(): Test {
  // not ok (property foo not found) -- flow is silent
  return (x) => parseInt(x)
}

function ex5(): Test {
  // not ok (function type is wrong) -- flow is silent
  var f = (x) => { };
  f.foo = "foo";
  return f
}
```

**Comment by samwgoldman** [comment]
Regarding `ex3`, I believe the issue lies in the fact that `function(x) { }` is a literal (or fresh) value, but flow regards it as a non-literal object based on the goofy reason matching when flowing from `ObjT` to `ObjT`.

**Comment by samwgoldman** [comment]
I added some failing tests to cover subtyping issues in [a branch](https://github.com/samwgoldman/flow/commit/5135a1926b987ea2e9dc6da5d291d7e864b582eb), but haven't been able to track down the cause.

**Comment by samwgoldman** [comment]
Partially fixed in #459

**Comment by samwgoldman** [comment]
OK. I think I was mistaken with `ex4` and `ex5` because those examples are failing as expected now. #459 can completely close this issue now.
